contracts/base/EscrowFactory.sol

```diff
@@ -144,75 +151,35 @@ contract EscrowFactory is Pausable, TInitializable, EscrowFactoryInterface {
     /**
         @notice It initializes this escrow contract factory instance.
         @param settingsAddress the settings contract address.
-        @param escrowLibraryAddress the escrow contract address.
+        @param escrowLogicAddress the escrow contract address.
      */
-    function initialize(address settingsAddress, address escrowLibraryAddress)
+    function initialize(address settingsAddress, address escrowLogicAddress)
         public
         isNotInitialized()
     {
         require(settingsAddress.isContract(), "SETTINGS_MUST_BE_A_CONTRACT");
-        require(escrowLibraryAddress.isContract(), "ESCROW_LIB_MUST_BE_A_CONTRACT");
+        require(escrowLogicAddress.isContract(), "ESCROW_LOGIC_MUST_BE_CONTRACT");
 
         TInitializable._initialize();
         Pausable.initialize(msg.sender);
 
-        escrowLibrary = escrowLibraryAddress;
+        upgradeEscrowLogic(escrowLogicAddress);
         settings = SettingsInterface(settingsAddress);
     }
 
-    /** Internal Functions */
-
     /**
-        @notice Computes a salt value given a loan id and sender address.
-        @return the computed salt value.
+        @notice It sets defines the new logic to be used for all Escrow contracts.
+        @param newLogic the new Escrow logic implementation.
      */
-    function _computeEscrowSalt(uint256 loanID) internal view returns (bytes32) {
-        return keccak256(abi.encodePacked(msg.sender, loanID));
-    }
+    function upgradeEscrowLogic(address newLogic) public onlyPauser() isInitialized() {
```

Comment:
Please, add the function in the interface and make it external (maybe we will need an internal function).